File: Rest-Api-V3/orders-private/orders-working/orders-working.js
const axios = require('axios');
const crypto = require('crypto');
require('dotenv').config();

async function fetchWorkingOrders(marketCode, orderId = null, clientOrderId = null) {
    const apiKey = process.env.API_KEY;
    const secretKey = process.env.API_SECRET;
    const ts = new Date().toISOString().split('.')[0] + 'Z';
    const nonce = Date.now().toString();
    const method = "/v3/orders/working";
    const apiUrl = "api.ox.fun";

    let queryString = `marketCode=${marketCode}`;
    if (orderId) {
        queryString += `&orderId=${orderId}`;
    }
    if (clientOrderId) {
        queryString += `&clientOrderId=${clientOrderId}`;
    }

    const msgString = `${ts}\n${nonce}\nGET\n${apiUrl}\n${method}\n${queryString}`;

    const sign = crypto.createHmac('sha256', secretKey)
        .update(msgString)
        .digest('base64');

    const headers = {
        'Content-Type': 'application/json',
        'AccessKey': apiKey,
        'Timestamp': ts,
        'Signature': sign,
        'Nonce': nonce
    };

    try {
        const response = await axios.get(`https://${apiUrl}${method}?${queryString}`, { headers });
        const data = response.data;
        if (data.success) {
            return data;
        } else {
            console.log('Failed to fetch working orders');
        }
    } catch (error) {
        console.log('Error fetching working orders:', error);
    }
}

// Example usage
fetchWorkingOrders("BTC-USD-SWAP-LIN").then(orders => console.log(orders));

File: orders-private/orders-working/orders-working.py
import os
import requests
import hmac
import hashlib
import base64
import time
from dotenv import load_dotenv

load_dotenv()

def fetch_working_orders(market_code, order_id=None, client_order_id=None):
    api_key = os.getenv('API_KEY')
    secret_key = os.getenv('API_SECRET').encode('utf-8')
    ts = time.strftime('%Y-%m-%dT%H:%M:%S', time.gmtime())
    nonce = str(int(time.time() * 1000))
    method = "/v3/orders/working"
    api_url = "api.ox.fun"

    query_string = f"marketCode={market_code}"
    if order_id:
        query_string += f"&orderId={order_id}"
    if client_order_id:
        query_string += f"&clientOrderId={client_order_id}"

    msg_string = f"{ts}\n{nonce}\nGET\n{api_url}\n{method}\n{query_string}"

    sign = base64.b64encode(hmac.new(secret_key, msg_string.encode('utf-8'), hashlib.sha256).digest()).decode('utf-8')

    headers = {
        'Content-Type': 'application/json',
        'AccessKey': api_key,
        'Timestamp': ts,
        'Signature': sign,
        'Nonce': nonce
    }

    try:
        response = requests.get(f"https://{api_url}{method}?{query_string}", headers=headers)
        response.raise_for_status()        
        data = response.json()
        if data.get('success'):
            return data
        else:
            print('Failed to fetch working orders')
    except requests.exceptions.RequestException as error:
        print('Error fetching working orders:', error)

# Example usage
orders = fetch_working_orders("BTC-USD-SWAP-LIN", order_id=None, client_order_id=None)
print(orders)